python/beeai_framework/adapters/litellm/embedding.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2025 © BeeAI a Series of LF Projects, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+import os
+from abc import ABC
+from itertools import chain
+from typing import Any
+
+import litellm
+from litellm import aembedding
+from litellm.litellm_core_utils.get_supported_openai_params import get_supported_openai_params
+from litellm.types.utils import EmbeddingResponse
+
+from beeai_framework.backend import EmbeddingModel
+from beeai_framework.backend.types import EmbeddingModelInput, EmbeddingModelOutput, EmbeddingModelUsage
+from beeai_framework.context import RunContext
+from beeai_framework.logger import Logger
+
+logger = Logger(__name__)
+
+
+class LiteLLMEmbeddingModel(EmbeddingModel, ABC):
+    def __init__(
+        self,
+        model_id: str,
+        *,
+        provider_id: str,
+        **kwargs: dict[str, Any],
+    ) -> None:
+        self._settings: dict[str, Any] = kwargs
+        self._model_id = model_id
+        self._litellm_provider_id = provider_id
+        self.supported_params = get_supported_openai_params(model=self.model_id, custom_llm_provider=provider_id) or []
+        # drop any unsupported parameters that were passed in
+        litellm.drop_params = True
+        # disable LiteLLM caching in favor of our own
+        litellm.disable_cache()  # type: ignore [attr-defined]
+
+    @property
+    def model_id(self) -> str:
+        return self._model_id
+
+    @staticmethod
+    def litellm_debug(enable: bool = True) -> None:
+        litellm.set_verbose = enable  # type: ignore [attr-defined]
+        litellm.suppress_debug_info = not enable
+
+        litellm.suppress_debug_info = not enable
+        litellm.logging = enable
+
+        logger = logging.getLogger("LiteLLM")
+        logger.setLevel(logging.DEBUG if enable else logging.CRITICAL + 1)
```

Comment:
This is duplicated. Extract this to a function shared between `LiteLLMChatModel` and `LiteLLMEmbeddingModel`. It can be stored in `adapters/litellm/utils.py`/